src/pickchar.nut

```diff
@@ -1,66 +1,80 @@
-::pickChar <- function() {
-	update()
-	resetDrawTarget()
 
+::pickCharSettings <- {}
+
+::pickCharInitialize <- function() {
 	//Make ordered list of characters
-	local charlist = []
+	pickCharSettings.charlist <- []
 	foreach(key, i in game.characters) {
 		local newitem = [key, i[2], i[3]]
-		charlist.push(newitem)
+		pickCharSettings.charlist.push(newitem)
 	}
 
 	//Sort characters
 	//Squirrel tables are sorted by hash, so the list will appear
 	//randomized if this is not done
-	charlist.sort(function(a, b) {
+	pickCharSettings.charlist.sort(function(a, b) {
 		if(a[0] > b[0]) return 1
 		if(a[0] < b[0]) return -1
 		return 0
 	})
-
-	local didpick = false
-	local picktimer = 30
-	local charslot = 0
-
+	
+	pickCharSettings.didpick <- false
+	pickCharSettings.picktimer <- 30
+	pickCharSettings.charslot <- 0
+	
 	//Set cursor to current character
-	for(local i = 0; i < charlist.len(); i++) {
-		if(charlist[i][0] == game.playerChar) {
-			charslot = i
+	for(local i = 0; i < pickCharSettings.charlist.len(); i++) {
+		if(pickCharSettings.charlist[i][0] == game.playerChar) {
+			pickCharSettings.charslot = i
 			break
 		}
 	}
+}
 
-	//Function loop
-	while(!getcon("pause", "press") && picktimer > 0) {
-		setDrawTarget(gvScreen)
-		drawSprite(bgCharSel, 0, screenW() / 2, 0)
+::pickChar <- function() {
 
-		if(!didpick) {
-			//Move selection
-			if(getcon("right", "press")) charslot++
-			if(getcon("left", "press")) charslot--
-			if(charslot < 0) charslot = charlist.len() - 1
-			if(charslot >= charlist.len()) charslot = 0
+	local didpick = pickCharSettings.didpick
+	local charslot = pickCharSettings.charslot
+	local charlist = pickCharSettings.charlist
+	local picktimer = pickCharSettings.picktimer
+	
+	if (getcon("pause", "press") || picktimer <= 0) {
+		gvGameMode = gmOverworld
```

Comment:
I also made this return to `gmOverworld` instead of the pause menu, meaning that once you pick a character, you're back on the overworld. Previously, you were brought back to the pause menu on the overworld.